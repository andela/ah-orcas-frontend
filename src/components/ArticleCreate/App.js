import React from 'react';
import './App.scss';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { postArticle } from '../../actions/articles.action';
import { clearTags } from '../../actions/tags.action';
import BodyinputComponent from './InputComponents';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      description: '',
      body: '',
      loading: false,
      success: false,
      buttonDisabled: false,
      error: false,
      tags: null,

    };
    this.handlechange = this.handlechange.bind(this);
    this.cleardata = this.cleardata.bind(this);
    this.enablePublishButton = this.enablePublishButton.bind(this);
    this.submitArticle = this.submitArticle.bind(this);
    this.handleError = this.handleError.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const { tags } = nextProps;
    this.setState({
      tags,
    });
  }

  handlechange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
    this.enablePublishButton();
  }


  cleardata() {
    this.setState({
      title: '',
      description: '',
      body: '',
      loading: false,
      success: true,
      tags: [],
    });
  }


  submitArticle() {
    const { postarticle } = this.props;
    postarticle(this.state, this.cleardata, this.handleError, this.props);
    this.setState(
      {
        tags: [],
        loading: true,
      },
    );
  }

  enablePublishButton() {
    const { title, body } = this.state;
    if (title && body) {
      this.setState({
        buttonDisabled: false,
      });
    }
  }

  handleError() {
    this.setState({
      error: true,
      success: false,
      loading: false,
    });
  }


  render() {
    const { postarticle } = this.props;
    const { title, description, body, buttonDisabled, loading, tags, success, error } = this.state;
    return (
      <div className="form-input">
        <BodyinputComponent
          submitArticle={this.submitArticle}
          buttonDisabled={buttonDisabled}
          postArticle={postarticle}
          body={body}
          tags={tags}
          handlechange={this.handlechange}
          title={title}
          description={description}
          loading={loading}
          success={success}
          error={error}
        />
      </div>
    );
  }
}

App.propTypes = {
  postarticle: propTypes.func,
  tags: propTypes.array,
};

App.defaultProps = {
  postarticle: propTypes.func,
  tags: [],
};

function mapStateToProps(state) {
  return {
    tags: state.tags,
  };
}


function mapDispatchToProps(dispatch) {
  return {
    postarticle: bindActionCreators(postArticle, dispatch),
    cleartags: bindActionCreators(clearTags, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
