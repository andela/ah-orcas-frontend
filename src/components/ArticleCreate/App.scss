
$primary-color: #2ebaae;

.title-input {
  margin-right: 2%;
  margin-top: 2%;
  padding-left: 4px;

}

.description-input {
  margin-right: 3%;
  margin-top: 2px;
  padding-left: 3px;

}

.form-input {
  height: 100vh;
  margin: 1%;
  padding-bottom: 5%;
  width: 99%;
}

.publish-btn {
  border-radius: 17px;
  font-size: 2rem;
  margin-right: 26px;
  margin-top: 1%;
  width: 20%;
}

.right {
  float: right;
}

.tag {
  border: 1px solid;
  margin-left: 1.5%;
  margin-right: 2%;
  margin-top: 1%;
  width: 47%;
}

.back-btn {
  background: none;
  border: 0;
  color: $primary-color;
  font-size: 1.5rem;
  margin-left: 1em;
}

.body-input {
  padding-left: 16px;
  textarea {
    border-radius: 5px;
    font-size: 2rem;
    height: 400px;
    padding-left: 16px;
    width: 99%;
  }
}

.form-control {
  border: 1px solid;
}

.Alert {
  background-color: #2ebaae;
  margin-bottom: 4px;
  margin-top: 6%;
  width: 99%;
}

.article-create-page {
  margin: auto;
  padding-top: 3%;
  width: 75%;

}

.tags {
  margin-top: 0.5%;
}

// .description:hover {
//   box-shadow: 3px 3px 3px;
//   z-index: 1;
// }
