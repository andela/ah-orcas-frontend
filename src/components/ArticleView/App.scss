$primary-color: #f5f5f5;
$article-container: #ffffff;

.description {
  color: inherit;
  font-size: 1.5rem;
  font-stretch: extra-expanded;
  font-style: italic;
  font-weight: bold;
  text-align: center;
}

.control-btn {
  margin-right: 1%;
  padding: 2%;
}

.article-body {
  font-size: 1.5rem;
}

.article-container {
  background: $article-container;
  margin-bottom: 6em;
  margin-left: 12em;
  margin-top: 5em;
  width: 70%;
}

.average {
  margin-left: 54em;
}

.title {
  font-weight: 900;
  text-transform: capitalize;
}

.reportlink {
  float: left;
  font-size: 14px;
  margin-left: 0.9em;
}

.socialshare {
  margin-top: 5em;
}
