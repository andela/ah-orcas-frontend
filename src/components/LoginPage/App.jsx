/* eslint-disable max-len */
import '../style.scss';
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import loginUser from '../../actions/login.action';
import Alert from './alert';
import SocialLoginConnected from './SocialLogin';
import FormGroups from './form';
import { SignupLoginLink } from '../SignUpPage/registerComponent';

const image = require('../../assets/images/1505193005.jpg');

export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.hideErrorDisplay = this.hideErrorDisplay.bind(this);
    this.state = {
      email: '',
      password: '',
      errorVisiblity: true,
    };
  }

    handleChange = (event) => {
      this.setState({
        [event.target.id]: event.target.value,
      });
    }

    handleSubmit = (event) => {
      event.preventDefault();
      const { dispatch } = this.props;
      dispatch(loginUser(this.state));
    }

    hideErrorDisplay = () => {
      this.setState({ errorVisiblity: false });
    }

    render() {
      const { userData } = this.props;
      const { errorVisiblity } = this.state;
      const { email, password } = this.state;
      return (
        <section className="sign-in">
          <div className="container">
            <div className="signin-content col-md-10">
              <SignupLoginLink imageClass="signin-image" image={image} link="/register" />
              <div className="signin-form" />
              <div className="col-md-5">
                <h2 className="form-title">Sign In</h2>
                <SocialLoginUl />
                <span className="social-label">Or</span>
                <form method="POST" className="register-form" id="login-form">
                  {userData.errors && errorVisiblity
                    ? <Alert message={userData.responseData.data.errors.error[0]} errorVisiblity={this.hideErrorDisplay} /> : ''}
                  <FormGroups
                    email={email}
                    password={password}
                    handleChange={this.handleChange}
                    onSubmit={this.handleSubmit}
                  />
                  <FormGroup />
                </form>
              </div>
            </div>
          </div>
        </section>
      );
    }
}

export const SocialLoginUl = () => (
  <div className="social-login">
    <ul className="socials">
      <SocialLoginConnected />
    </ul>
  </div>
);

// solve code length
export const FormGroup = () => (
  <div className="form-group form-button">
    <input type="submit" name="signin" id="signin" className="form-submit" value="Log in" />
    <div className="forgot-password">
      <Link to="/register" className="btn btn-link">Forgot Password?</Link>
    </div>
  </div>
);

Login.propTypes = {
  dispatch: PropTypes.func.isRequired,
  userData: PropTypes.object,
};

Login.defaultProps = {
  // dispatch: PropTypes.func,
  userData: {},
};
// pass the login reducer for maping as props to the login component
const mapStateToProps = ({ login }) => ({
  userData: login,
});

export default connect(mapStateToProps)(Login);
