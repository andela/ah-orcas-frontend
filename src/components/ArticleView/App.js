import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import './App.scss';
import { viewSingleArticle } from '../../actions/articles.action';
import Body from './ArticleViewComponents';
import DeleteArticle from '../ArticleDelete/DeleteArticle';
import RateArticle from '../RateArticle/rateArticleComponent';
import AverageRate from '../RateArticle/averageRate';
import LikesDislikes from '../LikeDislike/likedislike';
import DeleteModal from './DeleteModal/App';

let USER = {};
USER = JSON.parse(localStorage.getItem('user'));
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
    };
  }

  async componentDidMount() {
    const { singleArticle } = await this.props;
    singleArticle(window.location.href.substr(window.location.href.lastIndexOf('/') + 1));
  }

  componentWillReceiveProps(nextProps) {
    const { results } = nextProps;
    const { author } = results;
    this.setState(
      {
        username: author.username,
      },
    );
  }

  render() {
    const { rates, results } = this.props;
    const { username } = this.state;
    return (
      <div
        className="article-container"
      >
        <p className="average">
Average Ratings
          <AverageRate rates={rates} />
        </p>
        <Body
          tags={results.tags}
          username={results.title}
          title={results.title}
          descion={results.description}
          body={results.body}
          slug={results.slug}
        />
        <DeleteModal title={results.title} data={results} />
        <br />
        <LikesDislikes />
        <RateArticle />
        <div className="reportlink">
          {USER && USER.username === username ? '' : (
            <a href="/article/report">Report Article</a>
          ) }
        </div>
        <br />
        <DeleteArticle author={results.author} data={results} />
        {/* Attributes have been borrowed / re-used from forgot password link on login page. */}

      </div>
    );
  }
}

App.propTypes = {
  results: propTypes.object,
  rates: propTypes.number.isRequired,
};

App.defaultProps = {
  results: {},
};


function mapStateToProps(state) {
  return {
    results: state.viewarticle,
    rates: state.rating.average_rating,
  };
}


function mapDispatchToProps(dispatch) {
  return {
    singleArticle: bindActionCreators(viewSingleArticle, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
