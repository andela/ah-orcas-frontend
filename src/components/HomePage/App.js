import React, { Component } from 'react';
import './App.scss';
import './Animation.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import propTypes from 'prop-types';
import { getAllArticles } from '../../actions/articles.action';
import ArticleCard, { MyLoader } from './containers';
import { USER_IMAGE } from '../../constants/articles';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 1,
    };

    this.handleLoadMore = this.handleLoadMore.bind(this);
    this.handleLoadPrev = this.handleLoadPrev.bind(this);
  }

  componentDidMount() {
    const { fetchAllArticles } = this.props;
    const { page } = this.state;
    fetchAllArticles(page);
  }

  handleLoadMore() {
    const { page } = this.state;
    const { fetchAllArticles, articles } = this.props;
    // if the page === page_size, there could be more articles
    if (articles.length === 12) {
      this.setState({ page: page + 1 });
    }
    fetchAllArticles(page + 1);
  }

  handleLoadPrev() {
    const { page } = this.state;
    const { fetchAllArticles } = this.props;
    if (page === 1) {
      this.setState({ page: 1 });
    } else {
      this.setState({ page: page - 1 });
    }
    fetchAllArticles(page - 1);
  }

  render() {
    const { page } = this.state;
    const { articles } = this.props;
    const newArticles = articles.map(article => (
      <ArticleCard
        key={article.id}
        slug={article.slug}
        title={article.title}
        article={article}
        username={article.author.username}
        timecreated={article.timecreated}
        description={article.description}
        userimage={USER_IMAGE}
        comments={article.comments}
        likes={article.likes}
        dislikes={article.dislikes}
      />
    ));
    return (
      <section className="blog-area section">
        <div className="container">
          {
            articles.length !== 0
              ? (
                <div>
                  <div className="row">
                    {newArticles}
                  </div>
                  <ul className="pagination pagination-lg">
                    {
                      page !== 1
                        ? (
                          <li className="page-item"><a className="page-link" href onClick={this.handleLoadPrev}>Prev</a></li>
                        )
                        : (
                          ''
                        )
                    }
                    {
                      articles.length >= 12
                        ? (
                          <li className="page-item"><a className="page-link" href onClick={this.handleLoadMore}>Next</a></li>
                        )
                        : (
                          ''
                        )
                    }
                  </ul>
                </div>
              )
              : (
                <div className="row">
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                </div>
              )
          }
        </div>
      </section>
    );
  }
}

App.propTypes = {
  fetchAllArticles: propTypes.func,
  articles: propTypes.array,
};

App.defaultProps = {
  fetchAllArticles: propTypes.func,
  articles: propTypes.array,
};

function mapStateToProps(state) {
  return {
    articles: state.articles,
  };
}


function mapDispatchToProps(dispatch) {
  return {
    fetchAllArticles: bindActionCreators(getAllArticles, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
