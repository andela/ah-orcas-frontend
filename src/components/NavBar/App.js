import React from 'react';
import './App.scss';
import 'bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

export const NavBarComponent = () => (
  <div className="nav-container">
    <div className="nav-bar">
      <div className="horz-center nav-container">
        <div className="floated logo">Authors Haven</div>
        <ul className="floated right nav-div nav-link">
          <li className="rela-inline">Home</li>
          <li className="rela-inline">Search</li>
          <li className="rela-inline">About</li>
          <li className="rela-inline">Contact Us</li>
        </ul>
      </div>
    </div>
  </div>
);

export const FooterComponent = () => (
  <div className="navbar-light footer">
    <span>Â© 2018 Authors Haven.</span>
  </div>);
