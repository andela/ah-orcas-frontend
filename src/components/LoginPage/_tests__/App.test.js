
import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../App';

function setup() {
  const props = {
    login: jest.fn(),
  };
  const enzymeWrapper = shallow(<Login {...props} />);
  return {
    props,
    enzymeWrapper,
  };
}

describe('login components', () => {
  describe('login', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper.find('#button').length).toBe(0);
    });
  });
});
