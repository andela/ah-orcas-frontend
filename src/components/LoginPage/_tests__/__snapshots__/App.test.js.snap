// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login components login renders form group 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormGroup
    handleSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="form-submit"
          id="signin"
          name="signin"
          onClick={[MockFunction]}
          type="submit"
          value="Log in"
        />,
        <div
          className="forgot-password"
        >
          <a
            className="btn btn-link"
            href="/resetPassword"
          >
            Forgot Password?
          </a>
        </div>,
      ],
      "className": "form-group form-button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-submit",
          "id": "signin",
          "name": "signin",
          "onClick": [MockFunction],
          "type": "submit",
          "value": "Log in",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="btn btn-link"
            href="/resetPassword"
          >
            Forgot Password?
          </a>,
          "className": "forgot-password",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Forgot Password?",
            "className": "btn btn-link",
            "href": "/resetPassword",
          },
          "ref": null,
          "rendered": "Forgot Password?",
          "type": "a",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-submit"
            id="signin"
            name="signin"
            onClick={[MockFunction]}
            type="submit"
            value="Log in"
          />,
          <div
            className="forgot-password"
          >
            <a
              className="btn btn-link"
              href="/resetPassword"
            >
              Forgot Password?
            </a>
          </div>,
        ],
        "className": "form-group form-button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-submit",
            "id": "signin",
            "name": "signin",
            "onClick": [MockFunction],
            "type": "submit",
            "value": "Log in",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="btn btn-link"
              href="/resetPassword"
            >
              Forgot Password?
            </a>,
            "className": "forgot-password",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Forgot Password?",
              "className": "btn btn-link",
              "href": "/resetPassword",
            },
            "ref": null,
            "rendered": "Forgot Password?",
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`login components login renders social login 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialLoginUl />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(SocialAuth) />,
      "className": "social-login-new",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(SocialAuth) />,
        "className": "social-login-new",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`login components login should render form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    userData={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="signin-content col-md-10"
        >
          <SignupLoginLink
            image="1505193005.jpg"
            imageClass="signin-image"
            link="/register"
            text="Click here to register"
          />
          <div
            className="signin-form"
          />
          <div
            className="col-md-5"
          >
            <h2
              className="form-title"
            >
              Sign In
            </h2>
            <form
              className="register-form"
              id="login-form"
              method="POST"
            >
              
              <FormGroups
                email=""
                handleChange={[Function]}
                password=""
              />
              <FormGroup
                handleSubmit={[Function]}
              />
            </form>
            <hr
              className="style-eight"
            />
            <SocialLoginUl />
          </div>
        </div>
      </div>,
      "className": "sign-in",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signin-content col-md-10"
        >
          <SignupLoginLink
            image="1505193005.jpg"
            imageClass="signin-image"
            link="/register"
            text="Click here to register"
          />
          <div
            className="signin-form"
          />
          <div
            className="col-md-5"
          >
            <h2
              className="form-title"
            >
              Sign In
            </h2>
            <form
              className="register-form"
              id="login-form"
              method="POST"
            >
              
              <FormGroups
                email=""
                handleChange={[Function]}
                password=""
              />
              <FormGroup
                handleSubmit={[Function]}
              />
            </form>
            <hr
              className="style-eight"
            />
            <SocialLoginUl />
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SignupLoginLink
              image="1505193005.jpg"
              imageClass="signin-image"
              link="/register"
              text="Click here to register"
            />,
            <div
              className="signin-form"
            />,
            <div
              className="col-md-5"
            >
              <h2
                className="form-title"
              >
                Sign In
              </h2>
              <form
                className="register-form"
                id="login-form"
                method="POST"
              >
                
                <FormGroups
                  email=""
                  handleChange={[Function]}
                  password=""
                />
                <FormGroup
                  handleSubmit={[Function]}
                />
              </form>
              <hr
                className="style-eight"
              />
              <SocialLoginUl />
            </div>,
          ],
          "className": "signin-content col-md-10",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "image": "1505193005.jpg",
              "imageClass": "signin-image",
              "link": "/register",
              "text": "Click here to register",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signin-form",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="form-title"
                >
                  Sign In
                </h2>,
                <form
                  className="register-form"
                  id="login-form"
                  method="POST"
                >
                  
                  <FormGroups
                    email=""
                    handleChange={[Function]}
                    password=""
                  />
                  <FormGroup
                    handleSubmit={[Function]}
                  />
                </form>,
                <hr
                  className="style-eight"
                />,
                <SocialLoginUl />,
              ],
              "className": "col-md-5",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign In",
                  "className": "form-title",
                },
                "ref": null,
                "rendered": "Sign In",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    Array [],
                    "",
                    <FormGroups
                      email=""
                      handleChange={[Function]}
                      password=""
                    />,
                    <FormGroup
                      handleSubmit={[Function]}
                    />,
                  ],
                  "className": "register-form",
                  "id": "login-form",
                  "method": "POST",
                },
                "ref": null,
                "rendered": Array [
                  "",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "email": "",
                      "handleChange": [Function],
                      "password": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "handleSubmit": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "style-eight",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="signin-content col-md-10"
          >
            <SignupLoginLink
              image="1505193005.jpg"
              imageClass="signin-image"
              link="/register"
              text="Click here to register"
            />
            <div
              className="signin-form"
            />
            <div
              className="col-md-5"
            >
              <h2
                className="form-title"
              >
                Sign In
              </h2>
              <form
                className="register-form"
                id="login-form"
                method="POST"
              >
                
                <FormGroups
                  email=""
                  handleChange={[Function]}
                  password=""
                />
                <FormGroup
                  handleSubmit={[Function]}
                />
              </form>
              <hr
                className="style-eight"
              />
              <SocialLoginUl />
            </div>
          </div>
        </div>,
        "className": "sign-in",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="signin-content col-md-10"
          >
            <SignupLoginLink
              image="1505193005.jpg"
              imageClass="signin-image"
              link="/register"
              text="Click here to register"
            />
            <div
              className="signin-form"
            />
            <div
              className="col-md-5"
            >
              <h2
                className="form-title"
              >
                Sign In
              </h2>
              <form
                className="register-form"
                id="login-form"
                method="POST"
              >
                
                <FormGroups
                  email=""
                  handleChange={[Function]}
                  password=""
                />
                <FormGroup
                  handleSubmit={[Function]}
                />
              </form>
              <hr
                className="style-eight"
              />
              <SocialLoginUl />
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SignupLoginLink
                image="1505193005.jpg"
                imageClass="signin-image"
                link="/register"
                text="Click here to register"
              />,
              <div
                className="signin-form"
              />,
              <div
                className="col-md-5"
              >
                <h2
                  className="form-title"
                >
                  Sign In
                </h2>
                <form
                  className="register-form"
                  id="login-form"
                  method="POST"
                >
                  
                  <FormGroups
                    email=""
                    handleChange={[Function]}
                    password=""
                  />
                  <FormGroup
                    handleSubmit={[Function]}
                  />
                </form>
                <hr
                  className="style-eight"
                />
                <SocialLoginUl />
              </div>,
            ],
            "className": "signin-content col-md-10",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "image": "1505193005.jpg",
                "imageClass": "signin-image",
                "link": "/register",
                "text": "Click here to register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "signin-form",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="form-title"
                  >
                    Sign In
                  </h2>,
                  <form
                    className="register-form"
                    id="login-form"
                    method="POST"
                  >
                    
                    <FormGroups
                      email=""
                      handleChange={[Function]}
                      password=""
                    />
                    <FormGroup
                      handleSubmit={[Function]}
                    />
                  </form>,
                  <hr
                    className="style-eight"
                  />,
                  <SocialLoginUl />,
                ],
                "className": "col-md-5",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign In",
                    "className": "form-title",
                  },
                  "ref": null,
                  "rendered": "Sign In",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      Array [],
                      "",
                      <FormGroups
                        email=""
                        handleChange={[Function]}
                        password=""
                      />,
                      <FormGroup
                        handleSubmit={[Function]}
                      />,
                    ],
                    "className": "register-form",
                    "id": "login-form",
                    "method": "POST",
                  },
                  "ref": null,
                  "rendered": Array [
                    "",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "email": "",
                        "handleChange": [Function],
                        "password": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "handleSubmit": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "form",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "style-eight",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
